server/controllers/applications.coffee

```diff
@@ -21,26 +22,34 @@ sendError = (res, err, code=500) ->
 
 
 updateController = (count, callback) ->
-    log.info "controller: update"
-    exec "npm -g update cozy-controller", (err, stdout, stderr) ->
-        if err or stderr
-            if count < 2
-                updateController count + 1, callback
-            else
-                callback "Error during controller update after #{count + 1} try: #{stderr}"
+    # Check if a new version is available
+    latest 'cozy-controller', (err, version) ->
```

Comment:
@poupotte I think we should check `err` here. If we have an error while fetching the latest version from npm, it's probably not a good time to try updating packages with npm.